client/my-sites/plans-features-main/hooks/use-eligibility-for-term-savings-price-display.ts

```diff
@@ -2,32 +2,52 @@ import { isEnabled } from '@automattic/calypso-config';
 import {
 	getPlanSlugForTermVariant,
 	PlanSlug,
+	TERMS_LIST,
 	URL_FRIENDLY_TERMS_MAPPING,
 	UrlFriendlyTermType,
 } from '@automattic/calypso-products';
 import { AddOns, Plans } from '@automattic/data-stores';
-import { GridPlan } from '@automattic/plans-grid-next';
+import { HiddenPlans, PlansIntent, usePlanTypesWithIntent } from '@automattic/plans-grid-next';
+import usePlansFromTypes from '@automattic/plans-grid-next/src/hooks/data-store/use-plans-from-types';
 import useLongerPlanTermDefaultExperiment from './experiments/use-longer-plan-term-default-experiment';
 import useCheckPlanAvailabilityForPurchase from './use-check-plan-availability-for-purchase';
 
 const useEligibilityForTermSavingsPriceDisplay = ( {
-	gridPlans,
+	hiddenPlans,
+	intent,
+	isSubdomainNotGenerated,
+	selectedPlan,
+	term,
 	displayedIntervals,
 	coupon,
 	siteId,
 	storageAddOns,
 	isInSignup,
 }: {
-	gridPlans: GridPlan[];
+	hiddenPlans?: HiddenPlans;
+	intent?: PlansIntent;
+	isSubdomainNotGenerated?: boolean;
+	selectedPlan?: PlanSlug;
+	term: ( typeof TERMS_LIST )[ number ];
 	displayedIntervals: UrlFriendlyTermType[];
 	storageAddOns: ( AddOns.AddOnMeta | null )[] | null;
 	coupon?: string;
 	siteId?: number | null;
 	isInSignup?: boolean;
 } ) => {
 	const longerPlanTermDefaultExperiment = useLongerPlanTermDefaultExperiment();
-	const planSlugs = gridPlans.map( ( { planSlug } ) => planSlug );
-	const planSlugsForAllDisplayedIntervals = planSlugs.flatMap(
+	const availablePlanSlugs = usePlansFromTypes( {
+		planTypes: usePlanTypesWithIntent( {
+			intent: 'default',
+			selectedPlan,
+			siteId,
+			hiddenPlans,
+			isSubdomainNotGenerated,
+		} ),
+		term,
+		intent,
+	} );
+	const planSlugsForAllDisplayedIntervals = availablePlanSlugs.flatMap(
```

Comment:
This does make the hook a lot more complicated, also indirectly linked to how we generate plan slugs in `useGridPlans`, which might be hard to keep aligned going forward.

I wonder if we relax the criteria a bit (of reflecting or not reflecting storage selection) might help? e.g. let's show it 
 by default, and not show it if the `experiment.isEligibleForTermSavings` is set. Simple. So we disconnect it from needing access to `useEligibilityForTermSavingsPriceDisplay`, which does a lot more than that. WDYT?

Reply:
Although, not sure if that might help after we conclude the experiment and we run into this problem again 🤔

Reply:
Hmm so yeah, I think what you have here might be the way forward. Happy to give it some more thought later and get back to you.

p.s. we'd need to use the actual `intent` passed through here (without a default value), and also export `usePlansFromTypes` to import normally (not from the `src` folder). if we stick with this.

Reply:
> I wonder if we relax the criteria a bit (of reflecting or not reflecting storage selection) might help? e.g. let's show it
by default, and not show it if the experiment.isEligibleForTermSavings is set. Simple. So we disconnect it from needing access to useEligibilityForTermSavingsPriceDisplay, which does a lot more than that. WDYT?

The more I look at the code and weigh the complexity we introduce with the functionality we gain, the more I'm leaning towards this 👍